### PR TITLE
#131: adding new Json function

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: src/main/java/com/exasol/adapter/sql/SqlFunctionScalarJsonValue.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        public Optional<SqlNode> getExpression() {
+            return this.expression;
+        }
+
+        @Override
+        public boolean equals(final Object object) {
+            if (this == object) {
+                return true;
+            }
+            if (object == null || getClass() != object.getClass()) {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/sql/SqlFunctionScalarJsonValue.java:102]: The methods are auto-generated by Intellij.

File: src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java [hunk 2]
```diff
@@ -420,12 +430,27 @@ private SqlNode parsePredicateBetween(final JsonObject exp) {
...
+        return new SqlPredicateIsJson(expression, typeConstraint, keyUniquenessConstraint);
+    }
+
+    private SqlNode parsePredicateIsNotJson(final JsonObject jsonExpression) {
+        final SqlNode expression = parseExpression(jsonExpression.getJsonObject(EXPRESSION));
+        final SqlPredicateIsJson.TypeConstraints typeConstraint = SqlPredicateIsJson.TypeConstraints
+                .valueOf(jsonExpression.getString("typeConstraint").toUpperCase());
+        final SqlPredicateIsJson.KeyUniquenessConstraint keyUniquenessConstraint = SqlPredicateIsJson.KeyUniquenessConstraint
+                .of(jsonExpression.getString("keyUniquenessConstraint"));
+        return new SqlPredicateIsNotJson(expression, typeConstraint, keyUniquenessConstraint);
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/request/parser/PushdownSqlParser.java:448]: Here maybe using single class (suggested below), the common parts of these two methods can be extracted to another function. And set the predicate variable accordingly in these two methods.

File: src/main/java/com/exasol/adapter/sql/SqlPredicateIsNotJson.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.exasol.adapter.sql;
+
+import com.exasol.adapter.AdapterException;
+
+/**
+ * This class represents the {@link Predicate#IS_NOT_JSON} predicate.
+ */
+public class SqlPredicateIsNotJson extends AbstractSqlPredicateJson {
+    public SqlPredicateIsNotJson(final SqlNode expression,
+                                 final AbstractSqlPredicateJson.TypeConstraints typeConstraint,
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/sql/SqlPredicateIsNotJson.java:10]: done

File: src/main/java/com/exasol/adapter/sql/SqlPredicateIsNotJson.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    public <R> R accept(final SqlNodeVisitor<R> visitor) throws AdapterException {
+        return visitor.visit(this);
+    }
+
+    @Override
+    public String toSimpleSql() {
+        return " IS NOT JSON ";
+    }
+}
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/sql/SqlPredicateIsNotJson.java:29]: I thought about it, but as we have one class per one function, I left them like that. We also have null/not null, equal/not equal and so on

**Pull request merged**